app/src/main/java/com/example/bookwormadventuresdeluxe2/MyProfileFragment.java

```diff
@@ -35,50 +40,76 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     {
         // Inflate the layout for this fragment
         view = inflater.inflate(R.layout.fragment_my_profile, container, false);
+        signOutButton = view.findViewById(R.id.profile_logout);
+        signOutButton.setOnClickListener(this);
+
         edit = view.findViewById(R.id.profile_edit);
+        edit.setOnClickListener(this);
+
         /* Set title */
         appHeaderText = view.findViewById(R.id.app_header_title);
         appHeaderText.setText(R.string.my_profile_title);
 
-        edit.setOnClickListener(this);
+        firebaseAuth = FirebaseAuth.getInstance();
         return view;
     }
 
     @Override
     public void onClick(View view)
     {
-        final View editInfo = LayoutInflater.from(this.getContext()).inflate(R.layout.edit_profile, null);
+        switch (view.getId())
```

Comment:
add default case and throw exception if it is not expected to fallthrough

Reply:
added.